Build error

**Describe the bug**
when i install and import DragSelect v3.03 or v3.0.4, it throws typing error
Error: node_modules/dragselect/dist/DragSelect.d.ts:767:40 - error TS2304: Cannot find name ‘E_1’.
767     publish: <T extends DSCallbackName<E_1>>(eventName: T | T[], data: DSPublishMappings<E>[T]) => void;

I've checked `DragSelect.d.ts` in `dist` folder and `E_1` is mentioned there only once

--- a/DragSelect/CHANGELOG.md
+++ b/DragSelect/CHANGELOG.md
@@ -1,3 +1,7 @@
+# 3.0.5
+
+- Fix type issue during build bug [#212](https://github.com/ThibaultJanBeyer/DragSelect/issues/212)
+
 # 3.0.4
 
 - Fix scrolling bug ([#208](https://github.com/ThibaultJanBeyer/DragSelect/issues/208)) thanks to [@dbougan](https://github.com/dbougan)

--- a/DragSelect/src/DragSelect.ts
+++ b/DragSelect/src/DragSelect.ts
@@ -41,7 +41,7 @@ import Selection from './modules/Selection'
 import SelectorArea from './modules/SelectorArea'
 import SettingsStore from './stores/SettingsStore'
 import {
-  type DSCallbackName,
+  type DSInternalEventName,
   type DSInputElement,
   type Settings,
   type Vect2,
@@ -58,7 +58,7 @@ import { DSDropZone } from './modules/DropZone'
 // Setup
 /// ///////////////////////////////////////////////////////////////////////////////////
 
-class DragSelect<E extends DSInputElement> {
+class DragSelect<E extends DSInputElement = DSInputElement> {
   /** used to skip all current Selection and dragNdrop functionality */
   public continue: boolean = false
   private PubSub: PubSub<E>
@@ -151,7 +151,7 @@ class DragSelect<E extends DSInputElement> {
     id?: number
   ) => this.PubSub.unsubscribe(eventName, callback, id)
   /** Publish events */
-  public publish = <T extends DSCallbackName<E>>(
+  public publish = <T extends keyof DSPublicPublish<E> | DSInternalEventName>(
     eventName: T | T[],
     data: DSPublishMappings<E>[T]
   ) => this.PubSub.publish(eventName, data)

--- a/DragSelect/src/methods/subscriberAliases.ts
+++ b/DragSelect/src/methods/subscriberAliases.ts
@@ -6,9 +6,6 @@ import { DSInteractionPublishEventData } from '../modules/Interaction'
 import { DSSelectedPublishEventData } from '../modules/SelectedSet'
 import { DSInputElement } from '../types'
 
-export type DSPublicPublishEventNames<E extends DSInputElement> =
-  keyof DSPublicPublish<E>
-
 export type DSPublicPublishAdditionalEventData<E extends DSInputElement> = {
   /** The dropZone element that the element was dropped into (or the mouse is currently hovering over) */
   dropTarget?: DSDropZone<E>
@@ -113,7 +110,7 @@ type MappingObj<
   E extends DSInputElement,
   K extends keyof UsedPublishMappings<E>,
 > = {
-  name: DSPublicPublishEventNames<E>
+  name: keyof DSPublicPublish<E>
   condition?: Condition<UsedPublishMappings<E>[K], E>
   extraData?: Condition<UsedPublishMappings<E>[K], E>
 }

--- a/DragSelect/src/modules/PubSub.ts
+++ b/DragSelect/src/modules/PubSub.ts
@@ -54,7 +54,7 @@ export default class PubSub<E extends DSInputElement> {
    */
   unsubscribe = <K extends keyof DSPublishMappings<E>>(
     eventName: K,
-    callback?: DSCallback<DSPublishMappings<E>[K]>,
+    callback?: DSCallback<DSPublishMappings<E>[K]> | null,
     id?: number
   ) => {
     const index =

--- a/DragSelect/src/types.ts
+++ b/DragSelect/src/types.ts
@@ -1,14 +1,35 @@
-import { DSPublicPublishAdditionalEventData, DSPublicPublishEventNames } from "./methods/subscriberAliases"
-import { DSAreaPublishEventData, DSAreaPublishEventNames } from "./modules/Area"
-import { DSInteractionPublishEventData, DSInteractionPublishEventNames } from "./modules/Interaction"
-import { DSSelectablePublishEventData, DSSelectablePublishEventNames } from "./modules/SelectableSet"
-import { DSSelectedPublishEventData, DSSelectedPublishEventNames } from "./modules/SelectedSet"
-import { DSKeyStorePublishEventData, DSKeyStorePublishEventNames } from "./stores/KeyStore"
-import { DSPointerStorePublishEventData, DSPointerStorePublishEventNames } from "./stores/PointerStore"
-import { DSSettingsPublishEventData, DSSettingsPublishEventNames } from "./stores/SettingsStore"
+import {
+  type DSPublicPublishAdditionalEventData,
+  type DSPublicPublish,
+} from './methods/subscriberAliases'
+import { DSAreaPublishEventData, DSAreaPublishEventNames } from './modules/Area'
+import {
+  DSInteractionPublishEventData,
+  DSInteractionPublishEventNames,
+} from './modules/Interaction'
+import {
+  DSSelectablePublishEventData,
+  DSSelectablePublishEventNames,
+} from './modules/SelectableSet'
+import {
+  DSSelectedPublishEventData,
+  DSSelectedPublishEventNames,
+} from './modules/SelectedSet'
+import {
+  DSKeyStorePublishEventData,
+  DSKeyStorePublishEventNames,
+} from './stores/KeyStore'
+import {
+  DSPointerStorePublishEventData,
+  DSPointerStorePublishEventNames,
+} from './stores/PointerStore'
+import {
+  DSSettingsPublishEventData,
+  DSSettingsPublishEventNames,
+} from './stores/SettingsStore'
 
 export type Vect2 = {
-  x: number,
+  x: number
   y: number
 }
 
@@ -53,7 +74,7 @@ export type Settings<E extends DSInputElement> = {
   dropZones?: DSInputDropZone<E>[]
   /** [=1] how much % of the item has to be inside the dropzone to be considered inside (0 = barely touching, 1 = completely inside) */
   dropInsideThreshold?: number
-  /** [=0] how much % of the zone does the pointer has to be in to be considered a target (0 = anywhere in the zone, max: 0.5 = has to point at the center of the zone) */ 
+  /** [=0] how much % of the zone does the pointer has to be in to be considered a target (0 = anywhere in the zone, max: 0.5 = has to point at the center of the zone) */
   dropTargetThreshold?: number
   /** [=false] Whether to use Pointer Events to replace traditional Mouse or Touch Events. Useful for tools like Google Blockly. */
   usePointerEvents?: boolean
@@ -83,19 +104,18 @@ export type Settings<E extends DSInputElement> = {
   dropZoneInsideClass?: string
 }
 
-export type DSCallbackObject<E extends DSInputElement> = 
-  Readonly<
-    Partial<
-      & DSSettingsPublishEventData<E>
-      & DSAreaPublishEventData<E>
-      & DSKeyStorePublishEventData
-      & DSPointerStorePublishEventData
-      & DSInteractionPublishEventData
-      & DSSelectablePublishEventData<E>
-      & DSSelectedPublishEventData<E>
-      & DSPublicPublishAdditionalEventData<E>
-    >
+export type DSCallbackObject<E extends DSInputElement> = Readonly<
+  Partial<
+    DSSettingsPublishEventData<E> &
+      DSAreaPublishEventData<E> &
+      DSKeyStorePublishEventData &
+      DSPointerStorePublishEventData &
+      DSInteractionPublishEventData &
+      DSSelectablePublishEventData<E> &
+      DSSelectedPublishEventData<E> &
+      DSPublicPublishAdditionalEventData<E>
   >
+>
 
 export type DSInputDropZone<E extends DSInputElement> = {
   /** can be any unique identifier of type string */
@@ -107,19 +127,24 @@ export type DSInputDropZone<E extends DSInputElement> = {
 }
 
 export type DSElementPos = {
-  x: number,
-  y: number,
-  w: number,
-  h: number,
-  r: number,
+  x: number
+  y: number
+  w: number
+  h: number
+  r: number
   b: number
 }
 
-export type DSEdges = Array<'top'|'bottom'|'left'|'right'|undefined>
-export type DSEdgesObj = {top:number,bottom:number,left:number,right:number}
+export type DSEdges = Array<'top' | 'bottom' | 'left' | 'right' | undefined>
+export type DSEdgesObj = {
+  top: number
+  bottom: number
+  left: number
+  right: number
+}
 
 /** area within which you can drag */
-export type DSArea = HTMLElement|SVGElement|Document
+export type DSArea = HTMLElement | SVGElement | Document
 
 /** area in which you can drag */
 export type DSSelectorArea = HTMLElement
@@ -131,28 +156,34 @@ export type DSInputElement = HTMLElement | SVGElement
 // export type DSElement = HTMLElement|SVGElement
 
 /** An array of keys that allows switching to the multi-select mode */
-export type DSMultiSelectKeys = Array<'Shift'|'Control'|'Meta'|string>
+export type DSMultiSelectKeys = Array<'Shift' | 'Control' | 'Meta' | string>
 
-export type DSEvent = KeyboardEvent|MouseEvent|PointerEvent|TouchEvent
+export type DSEvent = KeyboardEvent | MouseEvent | PointerEvent | TouchEvent
 
-export type DSInternalEventName = DSSettingsPublishEventNames|DSAreaPublishEventNames|DSKeyStorePublishEventNames|DSPointerStorePublishEventNames|DSInteractionPublishEventNames|DSSelectablePublishEventNames|DSSelectedPublishEventNames
-export type DSCallbackName<E extends DSInputElement> = DSPublicPublishEventNames<E>|DSInternalEventName
+export type DSInternalEventName =
+  | DSSettingsPublishEventNames
+  | DSAreaPublishEventNames
+  | DSKeyStorePublishEventNames
+  | DSPointerStorePublishEventNames
+  | DSInteractionPublishEventNames
+  | DSSelectablePublishEventNames
+  | DSSelectedPublishEventNames
 
 export type DSBoundingRectBase = {
-  top: number,
-  left: number,
-  bottom: number,
-  right: number,
+  top: number
+  left: number
+  bottom: number
+  right: number
 }
 
 export interface DSBoundingRect extends DSBoundingRectBase {
-  width: number,
+  width: number
   height: number
 }
 
 export type DSDragKeys = {
-  up: Array<string>,
-  down: Array<string>,
-  left: Array<string>,
+  up: Array<string>
+  down: Array<string>
+  left: Array<string>
   right: Array<string>
 }

--- a/www/docs/guided-examples/React.mdx
+++ b/www/docs/guided-examples/React.mdx
@@ -22,17 +22,19 @@ Feel free to copy+paste the code:
 
 ```tsx DragSelectContext.tsx
 import React, { createContext, useState, useEffect, useContext } from "react";
-import DragSelect from "dragselect";
+import DragSelect, { DSInputElement } from "dragselect";
 
 type ProviderProps = {
   children: React.ReactNode;
-  settings?: ConstructorParameters<typeof DragSelect>[0];
+  settings?: ConstructorParameters<typeof DragSelect<DSInputElement>>[0];
 };
 
-const Context = createContext<DragSelect | undefined>(undefined);
+const Context = createContext<DragSelect<DSInputElement> | undefined>(
+  undefined
+);
 
 function DragSelectProvider({ children, settings = {} }: ProviderProps) {
-  const [ds, setDS] = useState<DragSelect>();
+  const [ds, setDS] = useState<DragSelect<DSInputElement>>();
 
   useEffect(() => {
     setDS((prevState) => {
@@ -108,7 +110,7 @@ export { DragSelectProvider, useDragSelect };
 ```jsx MyComponent.tsx
 import React from "react";
 
-import { DragSelectProvider } from "../DragSelectContext";
+import { DragSelectProvider } from "./DragSelectContext";
 
 export const MyComponent = () => (
   // you can add initial settings by passing a settings object
@@ -121,34 +123,38 @@ export const MyComponent = () => (
 ## 3. Use it
 
 ```jsx SomeOtherComponentsThatNeedsDragSelect.tsx
-import React, { useEffect, useRef } from 'react'
-import { useDragSelect } from '../../../DragSelectContext'
+import React, { useEffect, useRef } from "react";
+import { useDragSelect } from "./DragSelectContext";
 
-const SomeOtherComponentsThatNeedsDragSelect = () => {
-  const ds = useDragSelect()
-  const inputEl = useRef(null)
+export const SomeOtherComponentsThatNeedsDragSelect = () => {
+  const ds = useDragSelect();
+  const inputEl = useRef(null);
 
   // adding a selectable element
   useEffect(() => {
-    const element = inputEl.current as unknown as HTMLElement
-    if (!element || !ds) return
-    ds.addSelectables(element)
-  }, [ds, inputEl])
+    const element = inputEl.current as unknown as HTMLElement;
+    if (!element || !ds) return;
+    ds.addSelectables(element);
+  }, [ds, inputEl]);
 
   // subscribing to a callback
   useEffect(() => {
-    const id = ds.subscribe('DS:end', (e) => {
+    if (!ds) return;
+    const id = ds.subscribe("DS:end", (e) => {
       // do something
-      console.log(e)
-    })
+      console.log(e);
+    });
 
-    return () => ds.unsubscribe('DS:end', null, id)
-  }, [])
+    return () => ds.unsubscribe("DS:end", null, id!);
+  }, [ds]);
 
   return (
-    <button ref={inputEl} aria-labelledby="Selectable">Selectable</button>
-  )
-}
+    <button ref={inputEl} aria-labelledby="Selectable">
+      Selectable
+    </button>
+  );
+};
+
 ```
 
 DragSelect also exports some helper types for type safety, i.e. `DSPubCallback` which you can pass in the subscriber for the callback type safety:

--- a/www/src/pages/licenses/index.tsx
+++ b/www/src/pages/licenses/index.tsx
@@ -78,10 +78,14 @@ export default function License(): JSX.Element {
         </p>
         <h3 className={styles.section__subTitle}>Team member</h3>
         <p>
-          Commercial licenses are priced per team member. A team member is an
-          individual person permitted to make modifications for your
-          applications that uses DragSelect, whether such person is your
-          employee or a consultant or contractor providing services to you.
+          The Commercial license "Startup" is priced per team member. A team
+          member is an individual person permitted to make modifications for
+          your applications that uses DragSelect, whether such person is your
+          employee or a consultant or contractor providing services to you. One
+          startup license includes 5 team members: this means you buy one
+          startup license per 5 team members. The enterprise license however has
+          no limits: this means you only buy the enterprise license once no
+          matter how big your company is.
         </p>
         <h3 className={styles.section__subTitle}>Free updates</h3>
         <p>

